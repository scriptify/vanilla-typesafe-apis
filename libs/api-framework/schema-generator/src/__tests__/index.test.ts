/* eslint-disable @typescript-eslint/no-var-requires */
import * as path from 'path';
import execute from '../generate-controller-schema';
jest.setTimeout(20000);

describe('generate-api-json-schemas', () => {
  it('should generate the correct schema from complex types', async () => {
    const consoleErrorSpy = jest.spyOn(console, 'error');

    await execute('api-framework-schema-generator', path.resolve('.'));

    expect(consoleErrorSpy).not.toHaveBeenCalled();

    const fixturesFolder = path.resolve(__dirname, '__fixtures__');

    const reqSchema = require(path.join(fixturesFolder, 'Fake.request.schema.json'));

    expect(reqSchema).toMatchSnapshot();
  });
});
