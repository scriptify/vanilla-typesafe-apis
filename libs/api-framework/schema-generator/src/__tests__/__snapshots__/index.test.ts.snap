// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-api-json-schemas should generate the correct schema from complex types 1`] = `
Object {
  "$schema": "http://json-schema.org/draft-07/schema#",
  "additionalProperties": false,
  "properties": Object {
    "comments": Object {
      "items": Object {
        "additionalProperties": false,
        "properties": Object {
          "message": Object {
            "type": "string",
          },
          "user": Object {
            "additionalProperties": false,
            "properties": Object {
              "avatar": Object {
                "additionalProperties": false,
                "properties": Object {
                  "formats": Object {
                    "additionalProperties": false,
                    "properties": Object {
                      "large": Object {
                        "type": "string",
                      },
                      "medium": Object {
                        "type": "string",
                      },
                      "small": Object {
                        "type": "string",
                      },
                      "thumbnail": Object {
                        "type": "string",
                      },
                    },
                    "required": Array [
                      "large",
                      "medium",
                      "small",
                      "thumbnail",
                    ],
                    "type": "object",
                  },
                  "url": Object {
                    "type": "string",
                  },
                },
                "required": Array [
                  "formats",
                  "url",
                ],
                "type": "object",
              },
              "email": Object {
                "type": "string",
              },
              "name": Object {
                "type": "string",
              },
            },
            "required": Array [
              "avatar",
              "email",
              "name",
            ],
            "type": "object",
          },
        },
        "required": Array [
          "message",
          "user",
        ],
        "type": "object",
      },
      "type": "array",
    },
    "message": Object {
      "type": "string",
    },
    "price": Object {
      "anyOf": Array [
        Object {
          "additionalProperties": false,
          "properties": Object {
            "amount": Object {
              "type": "number",
            },
            "currency": Object {
              "enum": Array [
                "EUR",
                "USD",
              ],
              "type": "string",
            },
          },
          "required": Array [
            "amount",
            "currency",
          ],
          "type": "object",
        },
        Object {
          "type": "null",
        },
      ],
    },
  },
  "required": Array [
    "comments",
    "message",
  ],
  "type": "object",
}
`;
